[AArch64] Mark AESD and AESE instructions as commutative.

davemgreen opened the pull request:
This come from https://discourse.llvm.org/t/combining-aes-and-xor-can-be-improved-further/77248.

These instructions start out with:
```
  XOR Vd, Vn
  <some complicated math>
```
The initial XOR means that they can be treated as commutative, removing some of the unnecessary mov's introduced during register allocation.

davemgreen commented:
@iucoen FYI - I didn't seem to be able to add you as a reviewer

labrinea reviewed (approved):
I think this is correct. Reading ArmARM I can see that both AESE and AESD seem to perform an exclusive OR on their arguments before applying shift-rows/sub-bytes on the result.

davemgreen committed:
[AArch64] Mark AESD and AESE as commutative.

This come from https://discourse.llvm.org/t/combining-aes-and-xor-can-be-improved-further/77248.

These instructions start out with:
  XOR Vd, Vn
  <some complicated math>
The initial XOR means that they can be treated as commutative, removing some of
the unnecessary mov's introduced during register allocation.

iucoen commented:
32 bit ARM NEON also has the same instruction. Should be applied [there](https://github.com/llvm/llvm-project/blob/main/llvm/lib/Target/ARM/ARMInstrNEON.td#L7365) as well.

davemgreen commented:
> 32 bit ARM NEON also has the same instruction. Should be applied [there](https://github.com/llvm/llvm-project/blob/main/llvm/lib/Target/ARM/ARMInstrNEON.td#L7365) as well.

Yeah thanks, will do. We want keep them as separated patches though, for the different backends.
